plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -419,17 +420,22 @@ private void updatePartitionStatisticsBatch(Table table, Map<String, Function<Pa
             columnStatisticsUpdates.add(new GlueColumnStatisticsProvider.PartitionStatisticsUpdate(partition, updatedColumnStatistics));
         });
 
-        // Update basic statistics
-        Future<BatchUpdatePartitionResult> updatePartitionResult = glueClient.batchUpdatePartitionAsync(new BatchUpdatePartitionRequest()
-                .withCatalogId(catalogId)
-                .withDatabaseName(table.getDatabaseName())
-                .withTableName(table.getTableName())
-                .withEntries(partitionUpdateRequests.build()));
+        List<List<BatchUpdatePartitionRequestEntry>> partitionUpdateRequestsPartitioned = Lists.partition(partitionUpdateRequests.build(), BATCH_UPDATE_PARTITION_MAX_PAGE_SIZE);
+        List<Future<BatchUpdatePartitionResult>> partitionUpdateRequestsFutures = new ArrayList<>();
+        partitionUpdateRequestsPartitioned.forEach(partitionUpdateRequestsPartition -> {
+            // Update basic statistics
+            partitionUpdateRequestsFutures.add(glueClient.batchUpdatePartitionAsync(new BatchUpdatePartitionRequest()
+                    .withCatalogId(catalogId)
+                    .withDatabaseName(table.getDatabaseName())
+                    .withTableName(table.getTableName())
+                    .withEntries(partitionUpdateRequestsPartition)));
+        });
+
         try {
             // Update column statistics
             columnStatisticsProvider.updatePartitionStatistics(columnStatisticsUpdates.build());
             // Don't block on the batch update call until the column statistics have finished updating
-            getFutureValue(updatePartitionResult);
+            partitionUpdateRequestsFutures.forEach(MoreFutures::getFutureValue);
```

Comment:
preexisting, but is it OK to send `updatePartitionStatistics` even before `batchUpdatePartitionAsync` is finished?
ie is there no validation on the Glue side?

Reply:
Good point. IDK I assume there can be a validation that e.g. NDV is >= number of rows.
On the other hand there could also be validation that when you call `batchUpdatePartitionAsync` that it is in line with column statistics already present in Glue.

Reply:
I clicked through fields in https://docs.aws.amazon.com/glue/latest/webapi/API_UpdateColumnStatisticsForPartition.html and they do not document any smart validation rules. Just that values need to be >=0